controllers/etcdcluster_controller.go

```diff
@@ -274,6 +494,27 @@ func initialClusterMembers(cluster *etcdv1alpha1.EtcdCluster) []etcdv1alpha1.Ini
 	return members
 }
 
+func initialClusterMembersFromMemberList(cluster *etcdv1alpha1.EtcdCluster, members []etcdclient.Member) []etcdv1alpha1.InitialClusterMember {
+	initialMembers := make([]etcdv1alpha1.InitialClusterMember, len(members))
+	for i, member := range members {
+		name, err := peerNameForMember(member)
+		if err != nil {
```

Comment:
would we ever realistically get here in the normal flow? or is it only possible if somebody plays bad with the peer resources?

Reply:
Yeah, it wouldn't normally happen. This would mean that you'd added a new member to the membership API with no name and no client URLs. Assuming etcd lets you do that at all.